docs/markdown/Python/python/pex.md

```diff
@@ -20,6 +20,61 @@ A Pex can be bundled into a single `.pex` file. This file, when executed, knows
 
 Check out [blog.pantsbuild.org/pants-pex-and-docker](https://blog.pantsbuild.org/pants-pex-and-docker/) for how this workflow gets even better when combined with Pants's Docker support!
 
+Building a Pex
+--------------
+
+You define a Pex using the [`pex_binary`](doc:reference-pex_binary) target type:
+
+```python path/to/mybinary/BUILD
+python_sources(name="lib")
+
+pex_binary(
+    name="bin",
+    dependencies=[":lib"],
+    execution_mode="venv",
+)
+```
+
+You then use the `package` goal to build the Pex, which will be output under the [dist/] directory.
+
+```shell
+$ pants package path/to/mybinary:bin
+```
+
+There are several fields you can set on a `pex_binary` to configure the layout, entry point and behavior of the resulting Pex.  See the [documentation](doc:reference-pex_binary) for details.
+
+
+Setting the target platforms for a Pex
+--------------------------------------
+
+By default, the `package` goal builds a Pex that runs on the local machine's architecture and OS, and on a locally found interpreter compatible with your code's [interpreter constraints](doc:python-interpreter-compatibility). However, you can also build a multiplatform Pex - one that will run on multiple architecture+OS+interpreter combinations.
```

Comment:
> local machine's architecture and OS

Is there an interaction with environments that makes this slightly more subtle? Potentially it builds one appropriate for the environment (local, docker or remote) in which it was built, rather than always the local machine?

Maybe it's worth both: tweaking this sentence, and mentioning environments in general as another way to cross-build a single-platform PEX?

Reply:
> and on a locally found interpreter compatible with your code's [interpreter constraints](doc:python-interpreter-compatibility).

I don't think this is true today.  Instead it is the version of python that ran pex aka pant's own python. See #19514

Reply:
@cburroughs the shebang can have ~0 to do with the contents of the PEX. The shebang is determined by the interpreter used to run Pex unless you override, which you always should for a multiplatform or foreign platform PEX. The contents of the PEX, however can come from other local interpreters or even platforms / complete_platforms which represent interpreters not present.

Reply:
IOW, there are multiple axes here:
+ shebang: only important if you run `./my.pex` and not if you run `/this/python my.pex`.
+ 3rdparty dep contents

Ideally the 1st is derived from the second, but that is not always possible for a multitude of reasons. Only the PEX creator knows where the Pythons are installed on the various machines they target.

Reply:
> the shebang can have ~0 to do with the contents of the PEX. 

Right!  *Pants* today does *not* use the interpreter constraints to select a shebang for you.

Reply:
> Right! _Pants_ today does _not_ use the interpreter constraints to select a shebang for you.

That's not quite true. Pants sets `--interpreter-constraint` when it runs Pex, and Pex uses that to influence the shebang.

Reply:
I think no one has a great handle on this. If you use ICs, then Pex collects all those interpreters and builds the PEX with them in parallel. Throw in some platforms and or complete platforms and more parallel builds in the mix (the platforms / complete platforms) are totally orthogonal to ICs at build time. Of all the interpreters used to do all that building, 1, effectively random, interpreter from that set determines the default shebang.

Then, at runtime, an IC will be used to select candidate interpreters to start trying to boot the PEX with.